sunpy/sun/models.py

```diff
@@ -56,12 +57,14 @@
            10.157, 5.566, 2.259, 0.4483, 0.1528,
            0.042, 0.00361, 1.99e-7]
 
-_d = {'mass': _mass, 'luminosity': _luminosity, 'temperature': _temperature, 'density': _density}
-interior = pandas.DataFrame(_d, index = _radius)
-interior.index.name = 'radius'
-interior.units = {'radius': con.radius, 'mass': con.mass, 'luminosity': con.luminosity,
+_d = {'radius': _radius, 'mass': _mass, 'luminosity': _luminosity, 'temperature': _temperature, 'density': _density}
+_units = {'radius': con.radius, 'mass': con.mass, 'luminosity': con.luminosity,
                   'temperature': Quantity(1e6, 'K'), 'density': Quantity(1, 'g cm**-3')}
+interior = QTable(_d)
 interior.source = 'Turck-Chieze et al. (1988)'
+for labels in _units:
+    interior[labels].unit = _units[labels]
+interior.add_index('radius')
```

Comment:
Can you tell me what this does?

Reply:
Lines 65 and 66 add units to each of the columns. 
Line 67 makes the 'radius' column the index (since the radius was the index in the DataFrames as well).

The QTable does not print out any differently.

Reply:
Ah ok, I was not aware that radius was the index. Thanks!

Reply:
I just saw that there is a way to create QTables directly from DataFrames. Should I use that instead of creating the QTables from scratch ??

Reply:
No. What I would suggest to do is check that your changes are the same as if you had done that.

Reply:
My bad, doing that doesn't solve the initial issue as we will get the warning when we create the pandas DataFrame.

Reply:
Okay, will check that.

Reply:
I looked into it and conversion by using the method actually has some issues like it ignores the index column so in interior the final table ends up not having 'radius' column. 
I ran the tests and they are running successfully, please tell me if there are any changes that I need to make.

Reply:
I think as it stands, you do not need to make any more changes from my point of view.

We will need to wait for others to have a look over this pull request and make comments.